### DISCUSSION TITLE
Infer() use cases

### DISCUSSION THREAD

**Discussion by lyuboxa** [aarondl/sqlboiler#1264] [Q&A] [answered]
Hello,

In what cases is Infer() safe to use? I stumbled upon a foot-gun where `Infer()` was used on a table which had columns with `NOT NULL` and no default. I'm unclear if `Infer()` is ever safe in situations like this, since it overrides the expected behaviour in the database. 

Appreciate the input.

Thanks
Lyubo

**Comment by stephenafamo** [comment, marked as answer]
Infer is a way for SQLBoiler to translate Go's "zero type" to what your DB could be expecting. 

If a column is "NOT NULL" with no default, `Infer` will pass the zero value of the column if not set. The reason is that your database is expecting a value to be set.

This is how `infer` is expected to work. If you prefer more control you can use "Whitelist" or "Blacklist".

**Reply by lyuboxa** [reply]
@stephenafamo thank you.

I did some experiments and unfortunately neither of these provide the level of safety checks I need. 
The use of the `null` package allows us to infer when a column is set or not. I imagine one way to coerce the use of the `null` package will be through the configuration file?

**Reply by stephenafamo** [reply]
Yes. You can overwrite any type for any column through configuration.

**Reply by lyuboxa** [reply]
Thank you!
